### PR TITLE
feed: add description

### LINKED ISSUE #6245: description is blank in user public feed
i.e.: https://francetelevisions.cartodb.com/u/francetvinfo/me

### PR DESCRIPTION
**Pull request by matallo**
close #6245 

this is in ded05

CR @javierarce @saleiva

### CODE REVIEW

File: app/views/admin/pages/user_feed.html.erb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 <%= content_for(:title) do %>CartoDB &middot; <%= @name %> public feed<% end %>
+<%= content_for(:description) do %>This is <%= @name %> profile. CartoDB transforms location intelligence into dynamic renderings enabling discovery of trends and patterns.<% end %>
```
Comment (saleiva) [inline app/views/admin/pages/user_feed.html.erb]: Description should be less cheesy.

I'd propose something like 
"XXX has published X public maps and X public datasets. Look at this page to see XXX's latest activity on CartoDB."

If having 0 public datasets or 0 public maps for example, I'd say
"XXXX has publsihed X public datasets"

If having both equals to 0, I'd say something like

"XXXX is using CartoDB to transform location intelligence into dynamic renderings enabling discovery of trends and patterns."

**Comment by saleiva** [comment]
mentioning @brittanymicek to see if she can help with the texts a bit more

**Comment by brittanymicek** [comment]
1) [Username] uses CartoDB to transform location intelligence into dynamic renderings that enable discovery of trends and patterns.

EX: Planet Labs uses CartoDB to transform location intelligence into dynamic renderings that enable discovery of trends and patterns.

2) [Username] has published [X] data-driven visualizations and [X] public datasets. View [Username's] CartoDB profile for their latest activity.

EX: Wabion has published 10 data-driven visualizations and 10 public datasets. View Wabion's CartoDB profile for their latest activity.

3) [Username] has published [X] public datasets. View [Username's] CartoDB profile for the latest activity.

EX: Appgeo has published 30 public datasets. View Appgeo's CartoDB profile for their latest activity.

**Comment by matallo** [comment]
updated former descriptions with new ones

CR @javierarce

**Comment by matallo** [comment]
bump

CR @gfiorav @MariaCheca

File: app/controllers/admin/pages_controller.rb [hunk 2]
```diff
@@ -133,6 +133,31 @@ def user_feed
                             CartoDB.path(self, 'public_user_feed_home') and return
       end
 
+      description = "#{@name}"
```
Comment (gfiorav) [inline app/controllers/admin/pages_controller.rb]: Why not just `description = @name`?

Comment (matallo) [inline app/controllers/admin/pages_controller.rb]: tried it and when testing in staging the variable `@name` showed the whole description in the view, I had to revert it to this (don't really know why)

can you further investigate or propose a better solution for this?

Comment (gfiorav) [inline app/controllers/admin/pages_controller.rb]: Yep, that's a weird one. I believe the right way is:

`description = @name.dup`

File: app/controllers/admin/pages_controller.rb [hunk 3]
```diff
@@ -224,16 +249,16 @@ def render_datasets(vis_list, user=nil)
 
     @datasets.compact!
 
-    description = "#{@name} has"
+    description = "#{@name}"
```
Comment (gfiorav) [inline app/controllers/admin/pages_controller.rb]: Same jazz over here

File: app/controllers/admin/pages_controller.rb [hunk 4]
```diff
@@ -133,6 +133,31 @@ def user_feed
...
+      description = "#{@name}"
+
+      # TODO: move to helper
+      if @maps_count == 0 && @tables_num == 0
+        description << " uses CartoDB to transform location intelligence into dynamic renderings that enable discovery of trends and patterns"
+      else
+        description << " has"
+
+        unless @maps_count == 0
+          description << " created #{@maps_count} map#{@maps_count == 1 ? '' : 's'}"
```
Comment (gfiorav) [inline app/controllers/admin/pages_controller.rb]: Using `require 'active_support/inflector'`

You can explore things like 
```ruby
description << " created #{@maps_count} #{'map'.pluralize(@maps_count)}"
```

File: app/controllers/admin/pages_controller.rb [hunk 5]
```diff
@@ -133,6 +133,31 @@ def user_feed
...
+        end
+
+        unless @maps_count == 0 || @tables_num == 0
+          description << " and"
+        end
+
+        unless @tables_num == 0
+          description << " published #{@tables_num} public dataset#{@tables_num == 1 ? '' : 's'}"
```
Comment (gfiorav) [inline app/controllers/admin/pages_controller.rb]: Evaluate if
```ruby
description << " published #{@tables_num} #{'public dataset'.pluralize(@tables_num)}"
```

suits you better

File: app/controllers/admin/pages_controller.rb [hunk 6]
```diff
@@ -224,16 +249,16 @@ def render_datasets(vis_list, user=nil)
...
-    description = "#{@name} has"
+    description = "#{@name}"
 
     # TODO: move to helper
     if @datasets.size == 0
-      description << " not published any public dataset yet"
+      description << " uses CartoDB to transform location intelligence into dynamic renderings that enable discovery of trends and patterns"
     else
-      description << " published #{@datasets.size} public dataset#{@datasets.size == 1 ? "" : "s"}"
+      description << " has published #{@datasets.size} public dataset#{@datasets.size == 1 ? '' : 's'}"
```
Comment (gfiorav) [inline app/controllers/admin/pages_controller.rb]: Same here (for pluralising)

File: app/controllers/admin/pages_controller.rb [hunk 7]
```diff
@@ -257,14 +282,16 @@ def render_maps(vis_list, user=nil)
...
     # TODO: move to helper
     if @visualizations.size == 0 && @tables_num == 0
-      description << " not published any public dataset or map yet"
+      description << " uses CartoDB to transform location intelligence into dynamic renderings that enable discovery of trends and patterns"
     else
+      description << " has"
+
       unless @visualizations.size == 0
-        description << " created #{@visualizations.size} map#{@visualizations.size == 1 ? "" : "s"}"
+        description << " created #{@visualizations.size} map#{@visualizations.size == 1 ? '' : 's'}"
```
Comment (gfiorav) [inline app/controllers/admin/pages_controller.rb]: Same here (for pluralising)

File: app/controllers/admin/pages_controller.rb [hunk 8]
```diff
@@ -272,10 +299,10 @@ def render_maps(vis_list, user=nil)
       end
 
       unless @tables_num == 0
-        description << " published #{@tables_num} public dataset#{@tables_num == 1 ? "" : "s"}"
+        description << " published #{@tables_num} public dataset#{@tables_num == 1 ? '' : 's'}"
```
Comment (gfiorav) [inline app/controllers/admin/pages_controller.rb]: Same here (for pluralising)

**Comment by gfiorav** [comment]
@matallo check out the `.dup`comments.

Make use of the pluralize method if you want. 

After that, :+1:

**Pull request merged**